config/config.go

```diff
@@ -461,7 +503,17 @@ func (c *configStruct) SetProvider(provider string) error {
 
 	c.endpoints[c.SelectedEndpoint].Provider = provider
 	c.Provider = provider
-	WriteToFile()
+
+	// ignore errors like a barbarian
```

Comment:
You are wild

Reply:
But on a serious note, I guess you'll get a nil pointer reference below if this fails. I wouldn't think it would happen often, but maybe you should handle this just in case the config file somehow disappears before here

Reply:
You're right, checking `conf != nil` would be good here. I'll add it to be sure altough `readFromFile` [initializes the return value](https://github.com/giantswarm/gscliauth/blob/db5f140/config/config.go#L136) right at the start of the function.